docs: remove Firefox beta status, update firefoxGcInterval docs (#3168)
cypress-io/cypress-documentation: source/guides/references/configuration.md

## firefoxGcInterval

{% note warning %}

> The following section only applies if you are using a version of Firefox older than Firefox 80. `firefoxGcInterval` has no effect if you are using Firefox 80 or newer, since versions of Firefox newer than 80 do not have the bug. It is recommended to upgrade your version of Firefox to avoid this workaround.

Reviewer: "newer than 80" means 81+, but we want to say 80+.

> The following section only applies if you are using a version of Firefox older than Firefox 80. `firefoxGcInterval` has no effect if you are using Firefox 80 or newer, since versions of Firefox newer than 80 do not have the bug. It is recommended to upgrade your version of Firefox to avoid this workaround.

Author: ```suggestion
The following section only applies if you are using a version of Firefox older than Firefox 80. `firefoxGcInterval` has no effect if you are using Firefox 80 or newer, since the garbage collection bug was fixed in Firefox 80. It is recommended to upgrade your version of Firefox to avoid this workaround.
```
